src/go/k8s/controllers/cluster.redpanda.com/topic_controller.go

```diff
@@ -0,0 +1,634 @@
+		{map[string]string{}, kmsg.IncrementalAlterConfigOpSubtract}, // 3 == subtract
+	} {
+		for k, v := range pair.kvs {
+			config := kmsg.NewIncrementalAlterConfigsRequestResourceConfig()
+			config.Name = k
+			config.Op = pair.op
+			config.Value = kmsg.StringPtr(v)
+			configs = append(configs, config)
+		}
+	}
+
+	for keyToDelete := range deleteConf {
+		config := kmsg.NewIncrementalAlterConfigsRequestResourceConfig()
+		config.Name = keyToDelete
+		config.Op = kmsg.IncrementalAlterConfigOpDelete // 1 == delete
+		configs = append(configs, config)
+	}
+
+	if len(configs) == 0 {
+		if r.EventRecorder != nil {
+			r.EventRecorder.AnnotatedEventf(topic,
+				map[string]string{v2.GroupVersion.Group + "/revision": topic.ResourceVersion},
+				v1alpha1.EventTopicAlreadySynced, "Normal", "configuration not changed")
+		}
+		return nil
+	}
+
+	reqTopic := kmsg.NewIncrementalAlterConfigsRequestResource()
+	reqTopic.ResourceType = kmsg.ConfigResourceTypeTopic
+	reqTopic.ResourceName = topic.GetName()
+	reqTopic.Configs = configs
+	reqAltConfig.Resources = append(reqAltConfig.Resources, reqTopic)
+
+	l.V(TraceLevel).Info("alter topic configuration", "topic-name", topic.GetName(), "configs", configs)
+	respAltConfig, err := reqAltConfig.RequestWith(ctx, kafkaClient)
+	if err != nil {
+		if r.EventRecorder != nil {
+			r.EventRecorder.AnnotatedEventf(topic,
+				map[string]string{v2.GroupVersion.Group + "/revision": topic.ResourceVersion},
+				v1alpha1.EventTopicConfigurationAlteringFailure, "Error", "alter topic configuration (%s) library error: %s", topic.GetName(), err.Error())
+		}
+		return fmt.Errorf("alter topic configuration (%s) library error: %w", topic.GetName(), err)
+	}
+
+	if err = kerr.ErrorForCode(respAltConfig.Resources[0].ErrorCode); err != nil {
+		errMsg := NoneConstantString
+		if respAltConfig.Resources[0].ErrorMessage != nil {
+			errMsg = *respAltConfig.Resources[0].ErrorMessage
+		}
+		if r.EventRecorder != nil {
+			r.EventRecorder.AnnotatedEventf(topic,
+				map[string]string{v2.GroupVersion.Group + "/revision": topic.ResourceVersion},
+				v1alpha1.EventTopicConfigurationAlteringFailure, "Error", "alter topic configuration (%s) incremental alter config (%s): %s", topic.GetName(), errMsg, err.Error())
+		}
+		return fmt.Errorf("alter topic configuration (%s) incremental alter config (%s): %w", topic.GetName(), errMsg, err)
+	}
+	return nil
+}
+
+func (r *TopicReconciler) describeTopic(ctx context.Context, topic *v1alpha1.Topic, kafkaClient *kgo.Client) (*kmsg.DescribeConfigsResponse, error) {
+	req := kmsg.NewPtrDescribeConfigsRequest()
+	reqResource := kmsg.NewDescribeConfigsRequestResource()
+	reqResource.ResourceType = kmsg.ConfigResourceTypeTopic
+	reqResource.ResourceName = topic.GetName()
+	req.Resources = append(req.Resources, reqResource)
+	resp, err := req.RequestWith(ctx, kafkaClient)
+	if err != nil {
+		if r.EventRecorder != nil {
+			r.EventRecorder.AnnotatedEventf(topic,
```

Comment:
Would it be more convenient to move the eventRecoder updates into a function of its own?

Reply:
Sure, but event recorder from flux doesn't work nor event recorder from controller runtime manager. I mentioned it in our k8s sync meeting.

Reply:
? that seems to work for me. Ill check the code out to test. Make sure you create a new one for this controller.